### PR TITLE
Add arrays example

### PR DESCRIPTION
**Pull request by Jason-W123**

### CODE REVIEW

File: example_code/basic_examples/arrays/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+tokio = { version = "1.12.0", features = ["full"] }
+ethers = "2.0"
+eyre = "0.6.8"
+
+[features]
+export-abi = ["stylus-sdk/export-abi"]
+debug = ["stylus-sdk/debug"]
+
+[[bin]]
+name = "stylus-hello-world"
```
Comment (chrisco512) [inline example_code/basic_examples/arrays/Cargo.toml:25]: This should be "stylus-arrays-example" to match up with main.rs

File: src/app/basic_examples/arrays/page.mdx [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+tokio = { version = "1.12.0", features = ["full"] }
+ethers = "2.0"
+eyre = "0.6.8"
+
+[features]
+export-abi = ["stylus-sdk/export-abi"]
+debug = ["stylus-sdk/debug"]
+
+[[bin]]
+name = "stylus-hello-world"
```
Comment (chrisco512) [inline src/app/basic_examples/arrays/page.mdx:273]: Here as well

File: src/app/basic_examples/arrays/page.mdx [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        }
+        // if not found, return the size of arr
+        U256::from(self.arr3.len())
+    }
+}
+```
+
+# Full Example code:
+
+### src/main.rs
```
Comment (chrisco512) [inline src/app/basic_examples/arrays/page.mdx:147]: This should be `src/lib.rs`.

**Review by chrisco512 — APPROVED** [review]

File: example_code/applications/english_auction/Cargo.toml.bak [hunk 4]
```diff
@@ -0,0 +1,31 @@
+[package]
+name = "stylus-auction-example"
```
Comment (mahsamoosavi) [inline example_code/applications/english_auction/Cargo.toml.bak:2]:
```suggestion
name = "stylus-arrays-example"
```

File: src/app/basic_examples/arrays/page.mdx [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+    // set an element in the fixed length array
+    pub fn set_arr2_value(&mut self, index: U256, value: U256) {
+        self.arr2.setter(index).unwrap().set(value);
+    }
+}
+```
+Note that the fixed-size array can't push new elements.
+
+## Custome struct element array
```
Comment (mahsamoosavi) [inline src/app/basic_examples/arrays/page.mdx:91]:
```suggestion
## Custom struct element array
```

File: src/app/basic_examples/arrays/page.mdx [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+Custome struct element array is an array that can store custom struct elements.
```
Comment (mahsamoosavi) [inline src/app/basic_examples/arrays/page.mdx:92]:
```suggestion
Custom struct element array is an array that can store custom struct elements.
```

File: src/app/basic_examples/arrays/page.mdx [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+                return U256::from(i);
+            }
+        }
+        // if not found, return the size of arr
+        U256::from(self.arr3.len())
+    }
+}
+```
+
+# Full Example code:
```
Comment (mahsamoosavi) [inline src/app/basic_examples/arrays/page.mdx:145]:
```suggestion
# Full example code:
```

**Review by mahsamoosavi — CHANGES_REQUESTED** [review]
a few little edits and then LGTM

File: example_code/applications/english_auction/Cargo.toml.bak (hunk 4 above)
Comment (Jason-W123) [reply example_code/applications/english_auction/Cargo.toml.bak:2]: Oh, this one is in `English Auction`, so the current one is correct, we don't need to change, I will make a pr to change it back

**Review by mahsamoosavi — APPROVED** [review]

**Pull request merged**